### PR TITLE
fix(table): enable select all checkbox without totalRows, only data

### PR DESCRIPTION
**Pull request by mjlime**
## Review:
- [x] Commits are [atomic](https://seesparkbox.com/foundry/atomic_commits_with_git)
- [x] Commits have the correct *type* for the changes made
- [x] Commits with *breaking changes* are marked as such

### Browsers tested:
(Check any that applies, it's ok to leave boxes unchecked if testing something didn't seem relevant.)

Windows:
- [ ] Chrome
- [ ] Edge
- [ ] Firefox

Linux:
- [ ] Chrome
- [ ] Firefox

macOS:
- [x] Chrome
- [ ] Firefox
- [ ] Safari

Mobile:
- [ ] Chrome on Android
- [ ] iOS

### CODE REVIEW

File: src/components/table/table.tsx [hunk 1]
```diff
@@ -771,7 +771,7 @@ export class Table {
             >
                 <limel-checkbox
                     onChange={this.selectAllOnChange}
-                    disabled={!this.totalRows}
+                    disabled={!this.totalRows && !this.data.length}
```
Comment (adrianschmidt) [inline src/components/table/table.tsx:774]: Hmm… I wonder why we didn't just look at `this.data.length` in the first place? 🤔

Comment (adrianschmidt) [reply src/components/table/table.tsx:774]: This was implemented in #1632, fixing Lundalogik/crm-feature#2673. The intention is clearly to disable the checkbox is there are no rows currently being shown.

```suggestion
                    disabled={!this.data.length}
```

**Review by adrianschmidt — CHANGES_REQUESTED** [review]
With the suggested change, I'd approve and merge this 🙂

File: src/components/table/table.tsx (hunk 1 above)
Comment (mjlime) [reply src/components/table/table.tsx]: Done!

**Comment by mjlime** [comment]
> With the suggested change, I'd approve and merge this 🙂

Done!

**Review by adrianschmidt — APPROVED** [review]

**Pull request merged**